# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  779dfe07a4d90183c847ca3dc0c2b01dafc2f086
Misc: Don't backout of rev 85dcc90b5150: Don't show TB donation appeal.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1325,13 +1325,8 @@ pref("mail.useNewMailTabs", false);
 // Hide it and move the title to the center.
 pref("print.print_headerleft", "");
 pref("print.print_headercenter", "&T");
 pref("print.print_headerright", "");
 
 // Enable Masonry Layout for AddressBook.
 pref("layout.css.grid-template-masonry-value.enabled", true);
 
-// End of year donation appeal.
-pref("app.donation.eoy.version", 1);
-pref("app.donation.eoy.version.viewed", 0);
-pref("app.donation.eoy.url", "https://www.thunderbird.net/thunderbird/102.0/eoy/");
-
diff --git a/mail/base/content/msgMail3PaneWindow.js b/mail/base/content/msgMail3PaneWindow.js
--- a/mail/base/content/msgMail3PaneWindow.js
+++ b/mail/base/content/msgMail3PaneWindow.js
@@ -738,22 +738,16 @@ var gMailInit = {
       gSpacesToolbar.onLoad();
     }
 
     // All core modal dialogs are done, the user can now interact with the
     // 3-pane window. We need to notify this even if the user didn't setup any
     // mail account in order to trigger all the other areas of the application.
     Services.obs.notifyObservers(window, "mail-startup-done");
 
-    // Show the end of year donation appeal page.
-    if (this.shouldShowEOYDonationAppeal()) {
-      // Add a timeout to prevent opening the browser immediately at startup.
-      setTimeout(this.showEOYDonationAppeal, 2000);
-    }
-
     // Idle dispatch the telemetry reports.
     Services.tm.idleDispatchToMainThread(() => {
       reportAccountTypes();
       reportAddressBookTypes();
       reportAccountSizes();
       reportBooleanPreferences();
     });
   },
@@ -795,54 +789,16 @@ var gMailInit = {
     MailServices.mailSession.RemoveFolderListener(folderListener);
 
     // FIX ME - later we will be able to use onload from the overlay
     OnUnloadMsgHeaderPane();
 
     UnloadPanes();
     OnMailWindowUnload();
   },
-
-  /**
-   * Check if we can trigger the opening of the donation appeal page.
-   *
-   * @returns {boolean} - True if the donation appeal page should be opened.
-   */
-  shouldShowEOYDonationAppeal() {
-    let currentEOY = Services.prefs.getIntPref("app.donation.eoy.version", 1);
-    let viewedEOY = Services.prefs.getIntPref(
-      "app.donation.eoy.version.viewed",
-      0
-    );
-
-    // True if the user never saw the donation appeal, this is not a new
-    // profile (since users are already prompted to donate after downloading),
-    // and we're not running tests.
-    return (
-      viewedEOY < currentEOY &&
-      !specialTabs.shouldShowPolicyNotification() &&
-      !Cu.isInAutomation
-    );
-  },
-
-  /**
-   * Open the end of year appeal in a new web browser page. We don't open this
-   * in a tab due to the complexity of the donation site, and we don't want to
-   * handle that inside Thunderbird.
-   */
-  showEOYDonationAppeal() {
-    let url = Services.prefs.getStringPref("app.donation.eoy.url");
-    let messenger =
-      window.messenger ||
-      Cc["@mozilla.org/messenger;1"].createInstance(Ci.nsIMessenger);
-    messenger.launchExternalURL(url);
-
-    let currentEOY = Services.prefs.getIntPref("app.donation.eoy.version", 1);
-    Services.prefs.setIntPref("app.donation.eoy.version.viewed", currentEOY);
-  },
 };
 
 /**
  * Called at startup to verify if we have ny existing account, even if invalid,
  * and if not, it will trigger the Account Hub in a tab.
  *
  * @returns {boolean} - True if we have at least one existing account.
  */
